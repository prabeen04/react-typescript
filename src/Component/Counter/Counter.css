.counter-wrapper{
    height: 70vh;
    padding: 2rem;
}
.counter-display{
    display: flex;
    justify-content: center;
    margin-top: 150px;
}
.counter-display > h1{
    color: #aaa;
    font-family: Margarine;
    font-size: 70px;
}
.counter-controls{
    display: flex;
    justify-content: center;
}