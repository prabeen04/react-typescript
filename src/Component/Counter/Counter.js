import React, { useState } from 'react'

function Counter() {
    const [count, setCount] = useState(0)
    return (
        <div>
            <h1>{count}</h1>
        </div>
    )
}

export default Counter;