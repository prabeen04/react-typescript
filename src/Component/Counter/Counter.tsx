import * as React from 'react';
import './Counter.css';

const initialState = { count: 0 };
interface ICounterState {
    count: number
}
interface ICountProps {
}
const Counter: React.SFC<ICountProps> = (props: ICountProps) => {
    const [state, dispatch] = React.useReducer(appReducer, initialState)
    return (
        <div className='counter-wrapper' data-test='Counter'>
            <div className='counter-display'>
                <h1>{state.count}</h1>
            </div>
            <div className="counter-controls">
                <button className='btn btn-primary' onClick={() => dispatch({ type: 'INCREMENT' })}>Increase</button>&nbsp;
            <button className='btn btn-danger' onClick={() => dispatch({ type: 'DECREMENT' })}>Decrease</button>&nbsp;
            <button className='btn btn-warning' onClick={() => dispatch({ type: 'RESET' })}>Reset</button>&nbsp;
            <button className='btn btn-secondary' onClick={() => dispatch({ type: 'MULTIPLY', payload: 5 })}>Multiply by 5 </button>
            </div>
        </div>
    )
}

export default Counter;

const appReducer = (state: ICounterState, action: { type: string, payload?: number | null | undefined }) => {
    console.log(action)
    switch (action.type) {
        case 'INCREMENT':
            return { ...state, count: state.count + 1 }
        case 'DECREMENT':
            return { ...state, count: state.count - 1 }
        case 'MULTIPLY':
            return { ...state, count: state.count * 5 }
        case 'RESET':
            return { ...state, count: 0 }
        default:
            return { ...state }
    }
}