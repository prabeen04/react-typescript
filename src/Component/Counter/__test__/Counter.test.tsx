import * as React from 'react';
import Counter from '../Counter';
import { shallow } from "enzyme";

describe('<Counter/>', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Counter />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should render counter', () => {
        const wrapper = shallow(<Counter />)
        expect(wrapper.find('[data-test="Counter"]').length).toEqual(1)
    })
    it('should increment counter', () => {
        const wrapper = shallow(<Counter />)
        const incrementButton = wrapper.find('[data-test="increment-button"]')
        const counterDisplay = wrapper.find('[data-test="counter-display"]')
        expect(incrementButton.length).toEqual(1);
        // incrementButton.simulate('click');
        // incrementButton.props().      
        // expect(counterDisplay.text()).toEqual(1)
    })
})