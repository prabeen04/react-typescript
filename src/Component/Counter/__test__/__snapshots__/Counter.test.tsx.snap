// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="counter-display"
        >
          <h1>
            0
          </h1>
        </div>,
        <div
          className="counter-controls"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Increase
          </button>
           
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Decrease
          </button>
           
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            Reset
          </button>
           
          <button
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Multiply by 5 
          </button>
        </div>,
      ],
      "className": "counter-wrapper",
      "data-test": "Counter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            0
          </h1>,
          "className": "counter-display",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Increase
            </button>,
            " ",
            <button
              className="btn btn-danger"
              onClick={[Function]}
            >
              Decrease
            </button>,
            " ",
            <button
              className="btn btn-warning"
              onClick={[Function]}
            >
              Reset
            </button>,
            " ",
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              Multiply by 5 
            </button>,
          ],
          "className": "counter-controls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Increase",
              "className": "btn btn-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Increase",
            "type": "button",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Decrease",
              "className": "btn btn-danger",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Decrease",
            "type": "button",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset",
              "className": "btn btn-warning",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset",
            "type": "button",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Multiply by 5 ",
              "className": "btn btn-secondary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Multiply by 5 ",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="counter-display"
          >
            <h1>
              0
            </h1>
          </div>,
          <div
            className="counter-controls"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Increase
            </button>
             
            <button
              className="btn btn-danger"
              onClick={[Function]}
            >
              Decrease
            </button>
             
            <button
              className="btn btn-warning"
              onClick={[Function]}
            >
              Reset
            </button>
             
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              Multiply by 5 
            </button>
          </div>,
        ],
        "className": "counter-wrapper",
        "data-test": "Counter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              0
            </h1>,
            "className": "counter-display",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-primary"
                onClick={[Function]}
              >
                Increase
              </button>,
              " ",
              <button
                className="btn btn-danger"
                onClick={[Function]}
              >
                Decrease
              </button>,
              " ",
              <button
                className="btn btn-warning"
                onClick={[Function]}
              >
                Reset
              </button>,
              " ",
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                Multiply by 5 
              </button>,
            ],
            "className": "counter-controls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Increase",
                "className": "btn btn-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Increase",
              "type": "button",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Decrease",
                "className": "btn btn-danger",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Decrease",
              "type": "button",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset",
                "className": "btn btn-warning",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset",
              "type": "button",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Multiply by 5 ",
                "className": "btn btn-secondary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Multiply by 5 ",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
